src/Tochka.JsonRpc.Client/Models/BatchJsonRpcResult.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using JetBrains.Annotations;
+using Tochka.JsonRpc.Common;
+using Tochka.JsonRpc.Common.Models.Id;
+using Tochka.JsonRpc.Common.Models.Response;
+using Tochka.JsonRpc.Common.Models.Response.Errors;
+using Tochka.JsonRpc.Common.Models.Response.Untyped;
+
+namespace Tochka.JsonRpc.Client.Models;
+
+[PublicAPI]
+public class BatchJsonRpcResult : IBatchJsonRpcResult
+{
+    private readonly IJsonRpcCallContext context;
+    private readonly JsonSerializerOptions headersJsonSerializerOptions;
+    private readonly JsonSerializerOptions dataJsonSerializerOptions;
+    private readonly Dictionary<IRpcId, IResponse> responses;
+
+    public BatchJsonRpcResult(IJsonRpcCallContext context, JsonSerializerOptions headersJsonSerializerOptions, JsonSerializerOptions dataJsonSerializerOptions)
+    {
+        this.context = context ?? throw new ArgumentNullException(nameof(context));
```

Comment:
nullability say that IJsonRpcCallContext is not null by contract :)

Reply:
✅